src/internal/connector/graph/metadata_collection.go

```diff
@@ -0,0 +1,94 @@
+package graph
+
+import (
+	"bytes"
+	"context"
+	"io"
+
+	"github.com/alcionai/corso/src/internal/connector/support"
+	"github.com/alcionai/corso/src/internal/data"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+var (
+	_ data.Collection = &MetadataCollection{}
+	_ data.Stream     = &MetadataItem{}
+)
+
+type MetadataCollection struct {
```

Comment:
A comment outlining the difference in purpose between a MetadataCollection and the more standard "data" oriented collections would be helpful.